custom/icds_reports/management/commands/national_user_data_pull.py

```diff
@@ -0,0 +1,70 @@
+import csv
+
+from django.core.management.base import BaseCommand
+from corehq.apps.users.models import CommCareUser, WebUser
+from custom.icds_reports.const import INDIA_TIMEZONE
+from custom.icds_reports.models import ICDSAuditEntryRecord
+from django.db.models import Max
+from dimagi.utils.chunked import chunked
+
+
+class Command(BaseCommand):
+    help = "List of users with national level access on Dashboard along with usage"
+
+    def convert_to_ist(self, date):
+        if date is None:
+            return 'N/A'
+        date = date.astimezone(INDIA_TIMEZONE)
+        date_formatted = date.strftime("%d-%m-%Y")
+        return date_formatted
+
+    def _get_details(self, users, usernames, user_details):
+        """
+        :param users: user objects of all commcare and web users
+        :param usernames: username array eg. [username1, username2]
+        :param user_details: details dict contains created and role name eg.{username: [created_on, role_name]}
+        :return: usernames and user_details dict
+        """
+        for user in users:
+            role = user.get_role('icds-cas')
+            if (role in ('CPMU','Dashboard Only Access', 'TRP') or len(user.assigned_location_ids)==0) and user.has_permission('icds-cas', 'access_all_locations'):
+                usernames.append(user.username)
+                user_details.update({ user.username: [ user.created_on, role.name ]})
+        return usernames, user_details
+
+
+    def handle(self, *args, **options):
+        users = CommCareUser.by_domain('icds-cas')
+        web_users = WebUser.by_domain('icds-cas')
+        usernames = []
```

Comment:
its usually not a good idea to send a variable to a function to get updated. it will be good if the function just returns the values into it

Reply:
updated!